allauth/socialaccount/providers/fxa/provider.py

```diff
@@ -1,7 +1,42 @@
+from django.conf import settings
+
 from allauth.socialaccount import providers
 from allauth.socialaccount.providers.base import ProviderAccount
 from allauth.socialaccount.providers.oauth2.provider import OAuth2Provider
 
+# https://developer.mozilla.org/en-US/docs/Mozilla/Tech/Firefox_Accounts/Introduction#Firefox_Accounts_deployments
+_FXA_DEPLOYMENTS = {
```

Comment:
We should drop the pre-set, hard-coded `FXA_DEPLOYMENTS` values here and let the end user of `django-allauth` configure their `FXA_OAUTH_ENDPOINT` and `FXA_PROFILE_ENDPOINT` directly. For [push-dev-dashboard](https://github.com/mozilla-services/push-dev-dashboard), it made the [settings](https://github.com/mozilla-services/push-dev-dashboard/blob/4f1768527cd62191289c8c049cce511e553c3f46/dashboard/settings.py#L105-L112) and [view](https://github.com/mozilla-services/push-dev-dashboard/blob/4f1768527cd62191289c8c049cce511e553c3f46/dashboard/socialaccount/providers/fxa/views.py#L12-L28) code a lot smaller and cleaner...
```
SOCIALACCOUNT_PROVIDERS = {
    'fxa': {
        'OAUTH_ENDPOINT': config('FXA_OAUTH_ENDPOINT',
                                 'https://oauth-stable.dev.lcip.org/v1/'),
        'PROFILE_ENDPOINT': config('FXA_PROFILE_ENDPOINT',
                                   'https://stable.dev.lcip.org/profile/v1/profile')
   }
}
```
```
FXA_CONFIG = getattr(settings, 'SOCIALACCOUNT_PROVIDERS', {}).get('fxa', {})


class FirefoxAccountsOAuth2Adapter(BaseAdapter):
    provider_id = FirefoxAccountsProvider.id

    @property
    def access_token_url(self):
        return FXA_CONFIG.get('OAUTH_ENDPOINT', '') + '/token'

    @property
    def authorize_url(self):
        return FXA_CONFIG.get('OAUTH_ENDPOINT', '') + '/authorization'

    @property
    def profile_url(self):
        return FXA_CONFIG.get('PROFILE_ENDPOINT', '')
```